op-supervisor/supervisor/frontend/frontend.go

```diff
@@ -7,24 +7,34 @@ import (
 	"github.com/ethereum/go-ethereum/common/hexutil"
 
 	"github.com/ethereum-optimism/optimism/op-service/eth"
+	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
 	"github.com/ethereum-optimism/optimism/op-service/sources"
 	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
 )
 
+type FrontendMetrics interface {
+	opmetrics.RPCServerMetricer
+}
+
 type Backend interface {
 	sources.SupervisorAdminAPI
 	sources.SupervisorQueryAPI
 }
 
 type QueryFrontend struct {
 	Supervisor sources.SupervisorQueryAPI
+	Metrics    FrontendMetrics
 }
 
 var _ sources.SupervisorQueryAPI = (*QueryFrontend)(nil)
 
 // CheckMessage checks the safety-level of an individual message.
 // The payloadHash references the hash of the message-payload of the message.
 func (q *QueryFrontend) CheckMessage(ctx context.Context, identifier types.Identifier, payloadHash common.Hash, executingDescriptor types.ExecutingDescriptor) (types.SafetyLevel, error) {
+	if q.Metrics != nil {
+		done := q.Metrics.RecordRPCServerRequest("supervisor_checkMessage")
+		defer done()
+	}
```

Comment:
You could make this into a helper, like:

```golang
func (q *QueryFrontend)maybeRecordMetric(recordFn func()) {
  if q.Metrics != nil {
    recordFn()
  }
}
```

And then in the functions individually,
```golang
defer q.maybeRecordMetric(q.Metrics.RecordRPCServerRequest("..."))
```

By deferring the whole `if` clause it can get abstracted out.